File: cli/clients/cicd_hyper_a.ts
// SPDX-License-Identifier: PMPL-1.0

import { postJson } from "../http.ts";
import type { HttpOptions } from "../types.ts";

/** Publish a package to cicd-hyper-a registry */
export async function publish(
  path: string,
  baseUrl: string,
  token: string | undefined,
  opts: HttpOptions,
): Promise<void> {
  const body = JSON.stringify({ path });
  await postJson(baseUrl, "/publish", token, body, opts);
}

File: crates/opm-cli/src/clients/cicd_hyper_a.rs
// SPDX-License-Identifier: PMPL-1.0

use anyhow::Result;

use crate::clients::http;

pub fn publish(path: &str, base_url: &str, token: Option<&str>) -> Result<()> {
    let body = format!("{\"path\":\"{path}\"}");
    http::post_json(base_url, "/publish", token, &body)
}
